redwood-widget/src/iosMain/kotlin/app/cash/redwood/widget/UIViewChildren.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.widget
+
+import platform.UIKit.UIView
+import platform.UIKit.insertSubview
+import platform.UIKit.removeFromSuperview
+import platform.UIKit.setNeedsDisplay
+
+public class UIViewChildren(
+  private val parent: UIView,
+  private val insert: (UIView, Int) -> Unit = { view, index -> parent.insertSubview(view, index.toLong()) },
+) : Widget.Children<UIView> {
+  private val _widgets = MutableListChildren<UIView>()
+  public val widgets: List<Widget<UIView>> get() = _widgets
+
+  override fun insert(index: Int, widget: Widget<UIView>) {
+    _widgets.insert(index, widget)
+    insert(widget.value, index)
+  }
+
+  override fun move(fromIndex: Int, toIndex: Int, count: Int) {
+    _widgets.move(fromIndex, toIndex, count)
+
+    val subviews = Array(count) { offset ->
+      parent.typedSubviews[fromIndex + offset].also(UIView::removeFromSuperview)
+    }
```

Comment:
In the old one I always pulled from `fromIndex` (https://github.com/cashapp/redwood/blob/0a69b7b5af28d98aab61f26ed4d8ee1f4fd9b6c8/redwood-widget/src/iosMain/kotlin/app/cash/redwood/widget/UIViewChildren.kt#L33-L37) but maybe we never exercised it properly? Although I swear there are tests...

I'm still very wary of this implicit behavior of things not actually being removed from the collection and would prefer to do a second iteration pass or iterate backwards.

Or, even better, just ensure there's tests! I suspect there already are, actually.